### ISSUE TITLE
[Question] Does SyncedCachedEnforcer support LoadFilteredPolicy running concurrently with LoadPolicy?

### ISSUE THREAD

**Issue by longtengz** [casbin/casbin#1268]
**What's your scenario? What do you want to achieve?**

I'm trying to use `LoadFilteredPolicy` and `LoadPolicy` on SyncedCachedEnforcer in different places, thus they might be called in the same time. I think these two might interfere with each other, because sometimes my backend would wait on `LoadFilteredPolicy`, it's kind of random and does not have a particular pattern that I can debug with, so I think it's lock-related.

I'm using SyncedCachedEnforcer with this adapter https://github.com/pckhoi/casbin-pgx-adapter

**Comment by hsluoyz** [comment]
@PokIsemaine

**Comment by PokIsemaine** [comment]
@longtengz Thanks for your feedback, is your situation similar to using SyncCachedEnforcer's LoadPolicy and LoadFilteredPolicy in multiple goroutines at the same time?

It seems that the LoadFilteredPoloicy of the current SyncCachedEnforcer is not adapted to the SyncEnforcer to cause this problem.

![image](https://github.com/casbin/casbin/assets/46661603/0c39d221-8a10-400b-b779-087e12878044)
![image](https://github.com/casbin/casbin/assets/46661603/8dbae7ec-c713-4208-9b14-0a7f2748b449)

**Comment by PokIsemaine** [comment]
Ok, if you can provide a simple demo that can help me figure out what's wrong, thank you very much.

Also, is the `SyncEnforcer in the SyncCahcedEnforcer` you are using the same instance as the separate `SyncEnforcer`?

**Comment by PokIsemaine** [comment]
> I think these two might interfere with each other, because sometimes my backend would wait on `LoadFilteredPolicy`, it's kind of random and does not have a particular pattern that I can debug with, so I think it's lock-related.

So you don't think LoadPolicy and LoadFilteredPolicy should wait for each other? If so, why use SyncCachedEnforcer? Maybe you could try plain Enforcer or consider using different instances of SyncCachedEnforcer for LoadPolicy and LoadFilteredPolicy

**Comment by PokIsemaine** [comment]
So you don't think LoadPolicy and LoadFilteredPolicy should wait for each other? Maybe you could try plain Enforcer or consider using different instances of SyncCachedEnforcer for LoadPolicy and LoadFilteredPolicy

**Comment by PokIsemaine** [comment]
Ok, thanks for the explanation, I'll try to troubleshoot. 
This seems to take some time. It will help me a lot if you can provide a simple demo to help reproduce the problem

**Comment by hsluoyz** [comment]
@longtengz this issue cannot be stably reproduced. So unfortunately there's nothing much we can help

**Comment by hsluoyz** [comment]
@longtengz report in new issue.

Provide:

The code

What you expect

What you got
